### DISCUSSION TITLE
Could GDScript have a strict mode?

### DISCUSSION THREAD

**Discussion by blackears** [godotengine/godot-proposals#6677] [Scripting]
I was wondering how realistic it would be to have a strict mode for GDScript where the language behaved like a statically typed language instead of a dynamic one.  I come from a C++/C# background and personally find that the 'restrictions' statically typed languages impose are really helpful because they let you catch errors at compile time instead of runtime.  

GDScript has added a type system which I use, but it's still possible to do things like write code that tries to call a method that isn't there.

**Comment by dalexeev** [comment]
You can switch these and other warnings to Warn or Error:
![](https://user-images.githubusercontent.com/47700418/231098223-a5cd8284-d115-4192-be74-131c60a89af3.png)

**Reply by blackears** [reply]
I enabled these as warnings, but they don't seem to be warning me at all.  Even when I try to use methods and parameters that are not declared on classes.

**Reply by dalexeev** [reply]
Most warnings work, but some adjustments to trigger conditions are needed.
![](https://user-images.githubusercontent.com/47700418/231390500-d6efca9b-3072-459d-8efe-614676d41bb1.png)

**Comment by blackears** [comment]
Where do you find the debug settings?  I can't see them under either Project Settings or Editor Settings.

**Reply by AThousandShips** [reply]
Turn on Advanced settings

**Comment by Calinou** [comment]
- This is already being tracked in https://github.com/godotengine/godot-proposals/issues/173.
